### PR TITLE
Add a profile example with depends on.

### PR DESCRIPTION
**Pull request by Skarlso**

### CODE REVIEW

**Review by aclevername — DISMISSED** [review]
can we not just update the existing `weaveworks-nginx` profile?

**Comment by Skarlso** [comment]
@aclevername That would mess up a gazillion test cases in pctl. I like the separation of the example. Not trying to mash every single possibility into a single profile makes it more readable and understandable. While a complex all-in-one example is hard to read and decipher. WDYT?

**Comment by aclevername** [comment]
> @aclevername That would mess up a gazillion test cases in pctl. 

I feel like this is more of a smell that the tests are overloaded if this profile is getting tested more than 2 times

> I like the separation of the example. Not trying to mash every single possibility into a single profile makes it more readable and understandable. While a complex all-in-one example is hard to read and decipher. WDYT?

Wouldn't this big a quite a minor change to the existing test? Just checks for the creation of a `dependsOn` section in the kustomization, and the healthcheck right? So it only really adds 5-6 lines to an existing test? Or is there to the test than that.

I think a single profile that shows a lot of the functionality isn't necessarily a bad thing, I do get your concerns though. Definitely a balancing act.

**Comment by Skarlso** [comment]
@aclevername I must say, this does look nice, I guess. :D

**Review by aclevername — CHANGES_REQUESTED** [review]
hate to be this guy :see_no_evil: does it need a new artifact? Couldnt `nginx-chart` depend on `nginx-deployment`?

**Comment by Skarlso** [comment]
It does, because I want an artifact that depends on both a kustomize and a chart resource.

**Comment by Skarlso** [comment]
Also, once nested dependencies are available, nginx-deployment will depend on the nested profile.

File: weaveworks-nginx/profile.yaml [hunk 1]
```diff
@@ -24,3 +24,17 @@ spec:
...
+      chart:
+        url: https://charts.bitnami.com/bitnami
+        name: redis
+        version: "14.6.1"
+        defaultValues: |
+          master:
+            persistence:
+              enabled: false
+          replica:
+            replicaCount: 0
```
Comment (aclevername) [inline weaveworks-nginx/profile.yaml:37]: lol

**Review by aclevername — APPROVED** [review]
This needs to be merge same time as your PR right? Otherwise merging this will breaking existing tests (I think)

File: weaveworks-nginx/profile.yaml (hunk 1 above)
Comment (Skarlso) [reply weaveworks-nginx/profile.yaml:37]: Yeah this is needed for redis to properly start without any problems and extras. :D Also, we do need SOME diversity in the stuff we install. :DD

**Comment by Skarlso** [comment]
Yep. :)

**Comment by Skarlso** [comment]
PR which will use these and fixes the tests https://github.com/weaveworks/pctl/pull/220

**Pull request merged**